### PR TITLE
Update Oracle database EF API

### PR DESCRIPTION
**Pull request by sebastienros**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2195)

### CODE REVIEW

File: src/Components/Aspire.Oracle.EntityFrameworkCore/AspireOracleEFCoreExtensions.cs [hunk 1]
```diff
@@ -60,15 +50,117 @@ public static class AspireOracleEFCoreExtensions
...
+                    return optionsBuilder.Options;
+                },
+                oldDbContextOptionsDescriptor.Lifetime
+            );
+
+            builder.Services.Add(dbContextOptionsDescriptor);
+        }
+    }
+
+    private static OracleEntityFrameworkCoreSettings GetDbContextSettings<TContext>(IHostApplicationBuilder builder)
```
Comment (eerhardt) [inline src/Components/Aspire.Oracle.EntityFrameworkCore/AspireOracleEFCoreExtensions.cs:145]: (nit) can we put these methods in the same order across EF components? That way it is easy to diff across components and see what is different about them.

The SqlServer and Npgsql components have `ConfigureInstrumentation` above `GetDbContextSettings`.

File: tests/Aspire.Oracle.EntityFrameworkCore.Tests/ConformanceTests_Pooling.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 namespace Aspire.Oracle.EntityFrameworkCore.Tests;
 
-public class ConformanceTests_Pooling : ConformanceTests<TestDbContext, OracleEntityFrameworkCoreSettings>
+public class ConformanceTests : ConformanceTests<TestDbContext, OracleEntityFrameworkCoreSettings>
```
Comment (eerhardt) [inline tests/Aspire.Oracle.EntityFrameworkCore.Tests/ConformanceTests_Pooling.cs:15]: Can we rename the file too?

**Review by eerhardt — COMMENTED** [review]
Same comment as SqlServer about sharing common code across EF components.

File: src/Components/Aspire.Oracle.EntityFrameworkCore/AspireOracleEFCoreExtensions.cs (hunk 1 above)
Comment (sebastienros) [reply src/Components/Aspire.Oracle.EntityFrameworkCore/AspireOracleEFCoreExtensions.cs]: Done

**Comment by sebastienros** [comment]
> Same comment as SqlServer about sharing common code across EF components.

I'll do that when all PRs are merged https://github.com/dotnet/aspire/issues/2233

**Comment by eerhardt** [comment]
> I'll do that when all PRs are merged #2233

Why can't we do it on one of them - let's say SqlServer. Get that merged, and then update this one after that is merged?

**Comment by sebastienros** [comment]
Because I find it easier this way. Independent PRs, and then one more to refactor when they are in.

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**